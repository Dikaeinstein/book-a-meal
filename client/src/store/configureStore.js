import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import { createLogger } from 'redux-logger';
import throttle from 'lodash.throttle';
import { loadState, saveState } from '../helpers/localStorage';
import initialState from '../reducers/initialState';
import rootReducer from '../reducers';

/**
 * Return redux middleware
 */
const getMiddleware = () => {
  const logger = createLogger();

  const middleware =
    process.env.NODE_ENV !== 'production'
    && process.env.NODE_ENV !== 'test'
      ? applyMiddleware(thunk, logger)
      : applyMiddleware(thunk);

  return middleware;
};

/**
 * Configure the redux store
 *
 * @returns {object} Redux store
 */
const configureStore = () => {
  const persistedState = loadState();

  const state = persistedState
    ? { ...initialState, ...persistedState }
    : initialState;

  const store = createStore(
    rootReducer,
    state,
    compose(
      getMiddleware(),
      window.devToolsExtension ? window.devToolsExtension() : empty => empty,
    ),
  );

  // Throttle the saveState function to be called once every second
  store.subscribe(throttle(() => {
    saveState({ orders: store.getState().orders });
  }, 1000));

  return store;
};

export default configureStore;
