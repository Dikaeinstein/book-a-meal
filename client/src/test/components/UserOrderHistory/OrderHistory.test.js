import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import ConnectedOrderHistory, { OrderHistory }
  from '../../../components/UserOrderHistory/OrderHistory';


describe('<OrderHistory />', () => {
  const orders = [
    {
      id: 11,
      quantity: 1,
      total: '4000',
      status: 'delivered',
      expired: false,
      mealId: 3,
      userId: 4,
      meal: {
        id: 3,
        name: 'Amala and Ewedu',
        price: '4000',
      },
      user: {
        id: 4,
        name: 'Chinonso Okorie',
      },
    },
  ];
  it('should render with correct props', () => {
    const mockStore = configureStore();
    const store = mockStore({
      orders: {
        byId: { 11: { ...orders } },
        allIds: [11],
      },
    });
    const wrapper = shallow(<ConnectedOrderHistory store={store} />);
    expect(wrapper.props().orders.length).toBe(1);
  });
  it('should match snapshot', () => {
    const tree = shallow(<OrderHistory orders={orders} />);
    expect(tree).toMatchSnapshot();
  });
});
