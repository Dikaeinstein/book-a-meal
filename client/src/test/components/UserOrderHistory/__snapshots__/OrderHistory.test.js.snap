// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderHistory /> should match snapshot 1`] = `
<OrderList
  orders={
    Array [
      Object {
        "expired": false,
        "id": 11,
        "meal": Object {
          "id": 3,
          "name": "Amala and Ewedu",
          "price": "4000",
        },
        "mealId": 3,
        "quantity": 1,
        "status": "delivered",
        "total": "4000",
        "user": Object {
          "id": 4,
          "name": "Chinonso Okorie",
        },
        "userId": 4,
      },
    ]
  }
/>
`;
